### DISCUSSION TITLE
Packages for Linux distros

### DISCUSSION THREAD

**Discussion by stokito** [keycloak/keycloak#9643] [Ideas]
I would like to set up my own Keycloak instance. But all instructions that I found uses a manual installation by simply downloading an archive. There is an official [Docker image](https://hub.docker.com/r/jboss/keycloak/) but I still don't want to use it. The main reason is because Docker images are not automatically upgraded and I'm afraid that some zero day vulnerability will damage my users.
I found that there is some https://github.com/Palo-IT/keycloak-debian project template to build a Debian package. It's slightly outdated but it's sources seems easy to understand.
Is any intentions from Red Hat to create an RPM or maybe even a Debian repository with Keycloak?
I know that community can create such packages but still I think it would be better if the packages will be built and published by the Keycloak developers.
Or maybe at least you can accept a PR with `.spec` and `deb/control` files so that it will be simpler to build the DEB/RPM package from sources.

**Comment by thomasdarimont** [comment]
Having dedicated distribution deployables would be nice. However, I'd abstain from handcrafting such packages manually.

Instead, I'd take a look at https://jreleaser.org/ which supports packaging arbitrary java applications in various formats (also deb, .pkg, .dmg, .msi, .rpm, and many more...).

**Reply by aalmiray** [reply]
Happy to help out the team with advice and/or PR on using JReleaser 😉
